### PR TITLE
Fixes to timeout of sendMessageVector and refactor of sendBuffer for consistency

### PR DESCRIPTION
**Pull request by jasonpcarroll**
This PR fixes sendMessageVector improperly timing out and refactors both sendMessageVector and sendBuffer to be consistent with one another in their logic.

### CODE REVIEW

File: source/core_mqtt.c [hunk 1]
```diff
@@ -770,49 +772,65 @@ static int32_t sendMessageVector( MQTTContext_t * pContext,
...
-        /* Check for timeout. */
-        if( pContext->getTime() > timeoutTime )
-        {
-            break;
-        }
+    if( ( size_t ) bytesSentOrError == bytesToSend )
+    {
+        pContext->lastPacketTxTime = lastSendTimeMs;
+        LogDebug( ( "sendMessageVector: Successfully sent packet at time %lu.",
+                    ( unsigned long ) lastSendTimeMs ) );
```
Comment (AniruddhaKanhere) [inline source/core_mqtt.c:833]: Should this be done even when we are able to send even a partial packet?

File: source/core_mqtt.c [hunk 2]
```diff
@@ -733,35 +733,37 @@ static int32_t sendMessageVector( MQTTContext_t * pContext,
...
 {
-    uint32_t timeoutTime;
-    uint32_t bytesToSend = 0U;
-    int32_t bytesSentOrError = 0;
+    int32_t sendResult;
+    uint32_t lastSendTimeMs;
+    uint32_t timeSinceLastSendMs;
     TransportOutVector_t * pIoVectIterator;
     size_t vectorsToBeSent = ioVecCount;
+    size_t bytesToSend = 0;
```
Comment (AniruddhaKanhere) [inline source/core_mqtt.c:741]: There should a `U` at the end here. That is, it should be:
`size_t bytesToSend = 0U;`

**Review by AniruddhaKanhere — CHANGES_REQUESTED** [review]

File: source/core_mqtt.c [hunk 3]
```diff
@@ -770,49 +772,65 @@ static int32_t sendMessageVector( MQTTContext_t * pContext,
...
+
+            LogDebug( ( "sendMessageVector: Bytes Sent=%ld, Bytes Remaining=%lu",
+                        ( long int ) sendResult,
+                        ( unsigned long ) ( bytesToSend - ( size_t ) bytesSentOrError ) ) );
         }
-        else
+        else if( sendResult < 0 )
         {
             bytesSentOrError = sendResult;
+        }
```
[the remark this answers is not in the document]
Comment (Skptak) [reply source/core_mqtt.c:798]: Added in the empty else.

File: source/core_mqtt.c (hunk 2 above)
Comment (Skptak) [reply source/core_mqtt.c]: Added the ```U``` to this.

**Review by AniruddhaKanhere — CHANGES_REQUESTED** [review]

File: source/include/core_mqtt_config_defaults.h [hunk 4]
```diff
@@ -2,8 +2,6 @@
  * coreMQTT v2.0.0
  * Copyright (C) 2022 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
  *
- * SPDX-License-Identifier: MIT
- *
```
Comment (AniruddhaKanhere) [inline source/include/core_mqtt_config_defaults.h:6]: Can you add this back please?

**Review by AniruddhaKanhere — CHANGES_REQUESTED** [review]
Thank you for this PR @jasonpcarroll, can you make just one small change?

**Review by AniruddhaKanhere — APPROVED** [review]

**Review by Skptak — APPROVED** [review]

**Pull request merged**